src/Mod/Fem/App/FemPostPipeline.cpp

```diff
@@ -51,54 +59,209 @@
 using namespace Fem;
 using namespace App;
 
-PROPERTY_SOURCE(Fem::FemPostPipeline, Fem::FemPostObject)
-const char* FemPostPipeline::ModeEnums[] = {"Serial", "Parallel", "Custom", nullptr};
+
+vtkStandardNewMacro(FemFrameSourceAlgorithm);
+
+FemFrameSourceAlgorithm::FemFrameSourceAlgorithm::FemFrameSourceAlgorithm()
+{
+    // we are a source
+    SetNumberOfInputPorts(0);
+    SetNumberOfOutputPorts(1);
+}
+
+
+FemFrameSourceAlgorithm::FemFrameSourceAlgorithm::~FemFrameSourceAlgorithm()
+{}
+
+void FemFrameSourceAlgorithm::setDataObject(vtkSmartPointer<vtkDataObject> data)
+{
+    m_data = data;
+    Update();
+}
+
+bool FemFrameSourceAlgorithm::isValid()
+{
+    return m_data.GetPointer() != nullptr;
+}
+
+std::vector<double> FemFrameSourceAlgorithm::getFrameValues()
+{
+
+    // check if we have frame data
+    if (!m_data || !m_data->IsA("vtkMultiBlockDataSet")) {
+        return std::vector<double>();
+    }
+
+    // we have multiple frames! let's check the amount and times
+    vtkSmartPointer<vtkMultiBlockDataSet> multiblock = vtkMultiBlockDataSet::SafeDownCast(m_data);
+
+    unsigned long nblocks = multiblock->GetNumberOfBlocks();
+    std::vector<double> tFrames(nblocks);
+
+    for (unsigned long i = 0; i < nblocks; i++) {
+
+        vtkDataObject* block = multiblock->GetBlock(i);
+        // check if the TimeValue field is available
+        if (!block->GetFieldData()->HasArray("TimeValue")) {
+            break;
+        }
+
+        // store the time value!
+        vtkDataArray* TimeValue = block->GetFieldData()->GetArray("TimeValue");
+        if (!TimeValue->IsA("vtkFloatArray") || TimeValue->GetNumberOfTuples() < 1) {
+            break;
+        }
+
+        tFrames[i] = vtkFloatArray::SafeDownCast(TimeValue)->GetValue(0);
+    }
+
+    if (tFrames.size() != nblocks) {
+        // not every block had time data
+        return std::vector<double>();
+    }
```

Comment:
This is only true if for loop never break. Maybe we should return from inside the for loop then?